test/widgets/set_status_test.dart

```diff
@@ -0,0 +1,552 @@
+    check(findEmojiButton()).findsOne();
+    check(findStatusTextField()).findsOne();
+    checkSuggestionsVisible(suggestedUnicodeEmoji.keys.toList());
+  });
+
+  group('"Clear" & "Save" buttons', () {
+    group('initial state', () {
+      testWidgets('no status set -> buttons are disabled', (tester) async {
+        await setupPage(tester);
+
+        checkButtonEnabled(tester, clearButtonFinder, expected: false);
+        checkButtonEnabled(tester, saveButtonFinder, expected: false);
+      });
+
+      testWidgets('text & emoji are set -> "Clear" is enabled, "Save" is not', (tester) async {
+        await setupPage(tester, change: UserStatusChange(
+          text: OptionSome('Happy'),
+          emoji: OptionSome(StatusEmoji(emojiName: 'slight_smile',
+            emojiCode: '1f642', reactionType: ReactionType.unicodeEmoji))));
+
+        checkButtonEnabled(tester, clearButtonFinder, expected: true);
+        checkButtonEnabled(tester, saveButtonFinder, expected: false);
+      });
+
+      testWidgets('only text is set -> "Clear" is enabled, "Save" is not', (tester) async {
+        await setupPage(tester, change: UserStatusChange(
+          text: OptionSome('Happy'), emoji: OptionNone()));
+
+        checkButtonEnabled(tester, clearButtonFinder, expected: true);
+        checkButtonEnabled(tester, saveButtonFinder, expected: false);
+      });
+
+      testWidgets('only emoji is set -> "Clear" is enabled, "Save" is not', (tester) async {
+        await setupPage(tester, change: UserStatusChange(
+          text: OptionNone(),
+          emoji: OptionSome(StatusEmoji(emojiName: 'slight_smile',
+            emojiCode: '1f642', reactionType: ReactionType.unicodeEmoji))));
+
+        checkButtonEnabled(tester, clearButtonFinder, expected: true);
+        checkButtonEnabled(tester, saveButtonFinder, expected: false);
+      });
+    });
+
+    group('edit status', () {
+      Future<void> chooseEmojiFromPicker(WidgetTester tester, String code, {
+        bool emojiSelected = false,
+        required TestNavigatorObserver navObserver,
+        required ValueGetter<Route<dynamic>> currentRoute,
+      }) async {
+        await tester.tap(findEmojiButton(emojiSelected: emojiSelected));
+        check(currentRoute()).isNotNull().isA<ModalBottomSheetRoute<EmojiCandidate>>();
+        await navObserver.pumpPastTransition(tester);
+        // We use `find.descendant` to not match for an emoji in status
+        // suggestions in the underlying page.
+        await tester.tap(find.descendant(of: find.byType(EmojiPicker),
+          matching: find.text(tryParseEmojiCodeToUnicode(code)!)));
+        await tester.pump();
+        await navObserver.pumpPastTransition(tester);
+        check(currentRoute()).isNotNull().isA<MaterialAccountWidgetRoute>()
+          .page.isA<SetStatusPage>();
+      }
+
+      group('no status set, buttons are disabled', () {
+        testWidgets('emoji is added -> buttons are enabled', (tester) async {
```

Comment:
There are a lot of test cases in this group and the next one; and they feel pretty repetitive, totalling about 260 lines of code. It feels like a lot to read compared to the complexity of what it's testing, so I skimmed (until I got to the test cases for what the buttons do) rather than spend the time to read them all fully.

I don't think this area is critical to test, so let's not spend effort really polishing this set of tests; it's OK for them to go in while being somewhat repetitive and hard to read, because it'd be OK for this to go in without having most of this set of tests. But I'll leave one or two comments where I think it might be easy to cut them down.